src/main/java/de/hpi/bpt/argos/storage/hierarchy/HierarchyBuilderImpl.java

```diff
@@ -0,0 +1,89 @@
+package de.hpi.bpt.argos.storage.hierarchy;
+
+import de.hpi.bpt.argos.core.Argos;
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
+import de.hpi.bpt.argos.storage.dataModel.entity.Entity;
+import de.hpi.bpt.argos.storage.dataModel.entity.VirtualRoot;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public final class HierarchyBuilderImpl implements HierarchyBuilder {
+
+	private EntityHierarchyNode rootNode;
+
+	private static HierarchyBuilder instance;
+
+	/**
+	 * This constructor hides the default public one to implement the singleton pattern.
+	 */
+	private HierarchyBuilderImpl() {
+
+	}
+
+	/**
+	 * This method returns the singleton instance of this class.
+	 * @return - the singleton instance of this class
+	 */
+	public static HierarchyBuilder getInstance() {
+		if (instance == null) {
+			instance = new HierarchyBuilderImpl();
+		}
+
+		return instance;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public EntityHierarchyNode getEntityHierarchy() {
+		if (rootNode != null && !Argos.getTestMode()) {
```

Comment:
By the way, doesn't relate to the PR but I wanted to say that I don't like the name getTestMode either. Wouldn't expect it to return a boolean to be honest.